modin/core/execution/ray/generic/io/io.py

```diff
@@ -118,9 +118,23 @@ def to_csv(cls, qc, **kwargs):
         if not cls._to_csv_check_support(kwargs):
             return BaseIO.to_csv(qc, **kwargs)
 
-        # The partition id will be added to the queue, for which the moment
-        # of writing to the file has come
-        queue = Queue(maxsize=1)
+        @ray.remote
+        class SignalActor:  # pragma: no cover
+            def __init__(self, event_count):
+                self.events = [asyncio.Event() for _ in range(event_count)]
+
+            def send(self, event_idx):
+                self.events[event_idx].set()
+
+            async def wait(self, event_idx, should_wait=True):
+                if should_wait:
+                    await self.events[event_idx].wait()
+
+            def is_set(self, event_idx):
+                return self.events[event_idx].is_set()
+
+        signal_count = len(qc._modin_frame._partitions)
+        signals = SignalActor.remote(signal_count + 1)
```

Comment:
```suggestion
        signals = SignalActor.remote(len(qc._modin_frame._partitions) + 1)
```
what is the point of declaring a variable to be used once in next line only?